Doc/library/datetime.rst

```diff
@@ -542,6 +542,39 @@ Other constructors, all class methods:
 
    .. versionadded:: 3.8
 
+.. classmethod:: date.strptime(date_string, format)
+
+   Return a :class:`.date` corresponding to *date_string*, parsed according to
+   *format*. This is equivalent to::
+
+     date(*(time.strptime(date_string, format)[0:3]))
+
+   :exc:`ValueError` is raised if the date_string and format
+   can't be parsed by :func:`time.strptime` or if it returns a value which isn't a
+   time tuple.  See also :ref:`strftime-strptime-behavior` and
+   :meth:`date.fromisoformat`.
+
+   .. note
```

Comment:
Is it fine to have this note twice in the documentation?

Reply:
I think documentation experts will know better than me, but I think there's a way to define a substitution for this, like this:


```
|format-note|

.. |format-note| replace::
    If *format* specifies a day of month without a year a :exc:`DeprecationWarning` is emitted. This is to avoid a quadrenial leap year bug in code seeking to parse only a month and day, as the default year used when unspecified in the format is not a leap year. Such *format* values may raise an error in Python 3.15. The workaround is to always include a year in your *format*. If parsing *date_string* values that do not have a year, explicitly add a year that is a leap year before parsing:
    .. doctest::
        
        >> from datetime import date
        >> date_string = "02/29"
        >> when = date.strptime(f"{date_string};1984", "%m/%d;%Y")  # Avoids leap year bug.
        >> when.strftime("%B %d") # doctest: +SKIP
        'February 29'

.. versionadded:: 3.14
```

Then you put `|format-note|` in the other place as well.

Two other options:
  1. move this to a small fragment file (seems like overkill), and use an `..include` directive with it.
  2. put the note in one place, then link to it in the other place (e.g. "See the note on :func:`datetime.strptime`").

I don't think there's a simple way to do it with a parameter, though.

Reply:
I was just wondering if it was a problem it was mentioned twice. Do I link to `datetime.strptime` or is this fine?